cis194/week2/zhansongl/LogAnalysis.hs

```diff
@@ -0,0 +1,63 @@
+module LogAnalysis where
+
+import Log
+
+import Text.Megaparsec hiding (parse)
+import Text.Megaparsec.Char
+import qualified Text.Megaparsec.Char.Lexer as L
+import Data.Void
+
+type Parser = Parsec Void String
+
+messageTypeParser :: Parser MessageType
+messageTypeParser = do
+  charType <- oneOf "IEW"
+  case charType of
+       'I' -> return Info
+       'W' -> return Warning
+       'E' -> do 
+          space1
+          errorCode <- L.decimal
+          return (Error errorCode)
```

Comment:
The do notation is equivalent to the following:
`space1 *> L.decimal <&> Error`
`*>` disregards the result of `space1` and `<&>` is a flipped version of `<$>`

Reply:
hmm, it's more readable if you understand how it works, but less readable if you have no idea... will fix